addons/website_customer/controllers/main.py

```diff
@@ -85,15 +85,15 @@ def customers(self, country=None, industry=None, page=0, **post):
         countries = Partner.sudo().read_group(domain, ["id", "country_id"], groupby="country_id", orderby="country_id")
         country_count = Partner.sudo().search_count(domain)
 
+        fallback = False
         if country:
-            domain += [('country_id', '=', country.id)]
-            if country.id not in (x['country_id'][0] for x in countries if x['country_id']):
-                if country.exists():
-                    countries.append({
-                        'country_id_count': 0,
-                        'country_id': (country.id, country.name)
-                    })
-                    countries.sort(key=lambda d: (d['country_id'] or (0, ""))[1])
+            if country_count > 0 and country.exists() and country.id not in (x['country_id'][0] for x in countries if x['country_id']):
+                # fallback on all countries if no customer found for the country
+                # and there are matching customers for other countries
+                fallback = True
+                country = None
+            else:
+                domain += [('country_id', '=', country.id)]
 
         countries.insert(0, {
```

Comment:
This one needs to be made 'active' if country is none, otherwise the 'country' dropdown is empty when there's no result.

Reply:
If there is no result, the country selector will only have "All countries" in it because it shows the countries where there is at least one result.

Reply:
Yes but I mean 'active': True should be in the values, same as in /partners
Otherwise the selector won't show 'All Countries', because it's not selected. As shown below

![image](https://github.com/odoo/odoo/assets/43653136/d7f94f64-87b6-48fb-8a3b-33eaafae465f)

Reply:
Indeed :+1: Thanks it's fixed.